lib/ansible/module_utils/basic.py

```diff
@@ -1599,7 +1601,7 @@ def _log_invocation(self):
                 if not isinstance(param_val, basestring):
                     param_val = str(param_val)
                 elif isinstance(param_val, unicode):
-                    param_val = param_val.encode('utf-8')
+                    param_val = str(param_val.encode('utf-8'))
```

Comment:
In fact, the problem is that on py3, this creates a bytes object, and so I need to convert it to string again. So i guess the right fix is to not encode in the first place for python3 or something like this.